docs/reference/extensibility/NuGet-Cross-Platform-Plugins.md

```diff
@@ -0,0 +1,287 @@
+---
+title: NuGet cross platform plugins
+description: NuGet cross platform plugins for NuGet.exe, dotnet.exe, msbuild.exe and Visual Studio
+author: nkolev92
+ms.author: nikolev
+manager: rrelyea
+ms.date: 07/01/2018
+ms.topic: conceptual
+---
+
+# NuGet cross platform plugins
+
+In NuGet 4.8+ support for cross platform plugins has been added.
+This was achieved with by building a new plugin extensibility model, that has to conform to a strict set of rules of operation.
+The plugins are self-contained executables (runnables in the .NET Core world), that the NuGet Clients launch in a separate process.
+This is a true write once, run everywhere plugin. It will work with all NuGet client tools.
+The plugins can be either .NET Framework (NuGet.exe, MSBuild.exe and Visual Studio), or .NET Core (dotnet.exe).
+A versioned communication protocol between the NuGet Client and the plugin is defined. During the startup handshake, the 2 processes negotiate the protocol version.
+
+## How does it work
+
+The high level workflow can be described as follows:
+
+1. NuGet discovers available plugins.
+1. When applicable, NuGet will iterate over the plugins in priority order and starts them one by one.
+1. NuGet will use the first plugin that can service the request.
+1. The plugins will be shut down when they are no longer needed.
+
+## General plugin requirements
+
+The current protocol version is *2.0.0*.
+Under this version, the requirements are as follows:
+
+- Have a valid, trusted Authenticode signature assemblies that will run on Windows and Mono. There is no special trust requirement for assemblies run on Linux and Mac yet. [Relevant issue](https://github.com/NuGet/Home/issues/6702)
+- Support stateless launching under the current security context of NuGet client tools. For example, NuGet client tools will not perform elevation or additional initialization outside of the plugin protocol described later.
+- Be non interactive, unless explicitly specified.
+- Adhere to the negotiated plugin protocol version.
+- Respond to all requests within a reasonable time period.

```

Comment:
Are we imposing any timeout? worth to mention that here and if we aren't then we should....

Reply:
Yes, we do. 

There's a 5s request time out, but the plugin should send progress to prevent the request from timing out. 

I will call that out explicitly.

Reply:
Same as above, I've added a section covering this.